src/ascent/runtimes/flow_filters/ascent_runtime_vtkh_filters.cpp

```diff
@@ -208,9 +208,14 @@ VTKHMarchingCubes::execute()
     std::shared_ptr<VTKHCollection> collection = data_object->as_vtkh_collection();
 
     std::string field_name = params()["field"].as_string();
+
     if(!collection->has_field(field_name))
     {
-      detail::field_error(field_name, this->name(), collection);
+      bool throw_error = false;
+      detail::field_error(field_name, this->name(), collection, throw_error);
+      // this creates a data object with an invalid soource
+      set_output<DataObject>(new DataObject());
+      return;
```

Comment:
This is the basic idea of failing. Then we just have some special handling in the rendering.